test/jdk/java/nio/file/Files/ReadWriteString.java

```diff
@@ -304,6 +306,21 @@ public void testMalformedReadBytes(byte[] data, Charset csRead, Class<CharacterC
         throw new RuntimeException("An instance of " + expected + " should be thrown");
     }
 
+    // Verify File.readString with UTF16 to confirm proper string length and contents.
+    @Test
+    public void regression_8325590() throws IOException {
+        String original = "🤸";    // "\ud83e\udd38";
+        Path tmp = Files.createTempFile("tmp", ".txt");
+        Files.writeString(tmp, original, UTF_16);
+        String actual = Files.readString(tmp, UTF_16);
+        if (!actual.equals(original)) {
+            System.out.printf("expected (%s), was (%s)\n", original, actual);
+            System.out.printf("expected UTF_16 bytes: %s\n", Arrays.toString(original.getBytes(UTF_16)));
+            System.out.printf("actual UTF_16 bytes: %s\n", Arrays.toString(actual.getBytes(UTF_16)));
+        }
+        assertEquals(actual, original, "Round trip string mismatch with multi-byte encoding");
+    }
```

Comment:
The update to newStringNoRepl1 looks fine.

The added test is very different to the tests in this source file. We really need to expand the test to exercise a lot more charsets and input cases.  It's okay to have a targeted test for now but needs to be renamed to be consistent with the other tests. Also the other tests use testFiles as the file paths rather than putting files in /tmp.

Reply:
The UTF-16 charset is added to the existing `readString` test.
Additional charsets can be added, most will exercise the same code path in newStringNoRepl1 that uses a CharsetDecoder for all charsets other than UTF-8, ASCII, or ISO-8859-1.

The additional individual test is taken from the bug report and is not strictly necessary.